addon/components/x-select.hbs

```diff
@@ -17,38 +17,37 @@
       placeholder={{unless this.hasValues this.placeholder ""}}
       readonly={{unless this.canSearch "readonly"}}
       type="text"
-      oninput={{action "change" value="target.value"}}
-      onkeydown={{action "keypress"}}
-      onfocus={{action "focus"}}
-      onblur={{action "blur"}}
+      onkeydown={{this.keypress}}
+      onfocus={{this.focus}}
+      onblur={{this.blur}}
     >
   </div>
 
   {{#if this.canClear}}
-    <span class="es-clear-zone" title="Clear" aria-label="Clear" {{action "clear"}}>
+    <span class="es-clear-zone" title="Clear" aria-label="Clear" {{on "click" this.clear}}>
       <span class="es-clear">×</span>
     </span>
   {{/if}}
 
   {{#if this.hasDropdown}}
-    <span class="es-arrow-zone" {{action "dropdown" on="mouseUp" bubbles=false}}>
+    <span class="es-arrow-zone" {{on "click" this.onDropdown}}>
       <span class="es-arrow"></span>
     </span>
   {{/if}}
 </div>
 
 {{#if this.isOpen}}
   <div class="es-options">
-    {{component this.dropdown
-      parent=this
-      model=@model
-      token=this.token
-      values=@values
-      labelKey=this.labelKey
-      valueKey=this.valueKey
-      freeText=this.freeText
-      shouldFilter=this.shouldFilter
-      onSelect=(action "select")
-    }}
+    <this.dropdown
```

Comment:
This won't work with the current use of passing in strings, so it's a breaking change. Just want to call that out. This will be worthy of a version bump therefore.

Long-term, what do you think about passing in a block?

Reply:
Agree, although the documentation was already specifying `component` and not `string`.

Yes, updating to `@glimmer/component` & using blocks here would be really nice and it would cleanup a bit the intertwined components.